dbms/src/Storages/StorageFile.cpp

```diff
@@ -86,7 +87,8 @@ std::vector<std::string> LSWithRegexpMatching(const std::string & path_for_ls, c
         {
             if (re2::RE2::FullMatch(file_name, matcher))
             {
-                Strings result_part = LSWithRegexpMatching(full_path + "/", suffix_with_globs.substr(next_slash));
+                /// TODO: No recursion depth check. No protection for cyclic symlinks. It is a bug.
```

Comment:
Recursion depth is limited by pattern. "*" works only for depth = 1, for depth = 2 pattern path is "*/*". So we do not need additional check.

Reply:
Ok. There were no comments - it's a bug :)